From ecc0efa96e3c446586425d470657de7f2d5376bf Mon Sep 17 00:00:00 2001
From: Florence Blanc-Renaud <flo@redhat.com>
Date: Mon, 19 Jan 2026 17:00:41 +0100
Subject: [PATCH] ipatests: Fix xfail assertion for sssd 2.12.0

SSSD 2.12.0 provides fixes for
https://github.com/SSSD/sssd/issues/5989
https://github.com/SSSD/sssd/issues/7169

Update the condition expecting test failures in test_trust.py

Signed-off-by: Florence Blanc-Renaud <flo@redhat.com>
Reviewed-By: David Hanina <dhanina@redhat.com>
---
 ipatests/test_integration/test_trust.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/ipatests/test_integration/test_trust.py b/ipatests/test_integration/test_trust.py
index 13ad0afa4c1fb032d50f40cf7cb9b79283203225..0cab277c910a6d35f35b57e3068ee6f38706af59 100644
--- a/ipatests/test_integration/test_trust.py
+++ b/ipatests/test_integration/test_trust.py
@@ -1219,9 +1219,10 @@ class TestNonPosixAutoPrivateGroup(BaseTestTrust):
                 assert (uid == self.uid_override and gid == self.gid_override)
             test_group = self.clients[0].run_command(
                 ["id", nonposixuser]).stdout_text
-            cond2 = ((type == 'false'
-                      and sssd_version >= tasks.parse_version("2.9.4"))
-                     or type == 'hybrid')
+            cond2 = (((type == 'false'
+                       and sssd_version >= tasks.parse_version("2.9.4"))
+                      or type == 'hybrid')
+                     and sssd_version < tasks.parse_version("2.12.0"))
             with xfail_context(cond2,
                                'https://github.com/SSSD/sssd/issues/5989 '
                                'and 7169'):
@@ -1347,7 +1348,8 @@ class TestPosixAutoPrivateGroup(BaseTestTrust):
                     and gid == self.gid_override)
             result = self.clients[0].run_command(['id', posixuser])
             sssd_version = tasks.get_sssd_version(self.clients[0])
-            bad_version = sssd_version >= tasks.parse_version("2.9.4")
+            bad_version = (tasks.parse_version("2.9.4") <= sssd_version
+                           < tasks.parse_version("2.12.0"))
             with xfail_context(bad_version and type in ('false', 'hybrid'),
                  "https://github.com/SSSD/sssd/issues/7169"):
                 assert "10047(testgroup@{0})".format(
-- 
2.52.0

